src/Coin.sol

```diff
@@ -0,0 +1,47 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.0;
+
+import "lib/openzeppelin-contracts/contracts/token/ERC20/ERC20.sol";
+import "lib/openzeppelin-contracts/contracts/access/Ownable.sol";
+import "lib/openzeppelin-contracts/contracts/utils/math/SafeMath.sol";
+import "lib/chainlink/contracts/src/v0.8/interfaces/AggregatorV3Interface.sol";
+import "./BasketHandler.sol";
+import "./PriceConverter.sol";
+import "./Notary.sol";
+
+/**
+ * @dev Coin contract manages the supply of the stable coin.
+ *
+ * This contract is simple ER20 token contract with constraints on minting, where minting
+ * is limited to a Notary registered Position that is above the mininum collateralization
+ * ratio.
+ */
+contract Coin is ERC20 {
+    Notary notary;
+
+    constructor(address _notaryAddress) ERC20("Coin", "coin") {
+        notary = Notary(_notaryAddress);
+    }
+
+    /**
+     * @dev Mints for authenticated position contracts.
+     */
+
+    function mint(
+        address _positionAddress,
+        address _receiver,
+        uint256 _moreDebt
+    ) external {
+        // require(
+        //     notary.isValidPosition(_positionAddress),
+        //     "Caller is not authorized to mint"
+        // );
+        _mint(_receiver, _moreDebt);
+    }
```

Comment:
Anyone can call this minting function. Perhaps the following could fix it:

```suggestion
    /**
     * @dev Mints for authenticated position contracts.
     */
    function mint(address _receiver, uint256 _moreDebt) public {
        address _positionAddress = msg.sender;
        require(notary.isValidPosition(_positionAddress), "Caller is not authorized to mint");
        Position position = Position(_positionAddress);
        require(position.canTake(_moreDebt), "Position cannot take debt");
        _mint(_receiver, _moreDebt);
    }
```

Reply:
Agreed, only the Notary (or manager) and the position creator (debtor) should be able to mint. Notary should have all access and debtor will only be able to mint an amount depending on the collateral deposited.